src/neptune/new/attributes/atoms/file.py

```diff
@@ -34,16 +35,16 @@ class File(Atom):
     def assign(self, value: FileVal, wait: bool = False) -> None:
         verify_type("value", value, FileVal)
 
-        if value.path is not None:
+        if value.file_type is FileType.LOCAL_FILE:
             operation = UploadFile(self._path, ext=value.extension, file_path=os.path.abspath(value.path))
-        elif value.content is not None:
+        elif value.file_type is FileType.IN_MEMORY:
             operation = UploadFileContent(
                 self._path,
                 ext=value.extension,
                 file_content=base64_encode(value.content),
             )
         else:
-            raise ValueError("File path and content are None")
+            raise ValueError(f"Unexpected FileType: {value.file_type}")
```

Comment:
Is the message visible for clients? We would add to message `Supported types: {FileType.LOCAL_FILE} and FileType.IN_MEMORY`
Client first! :)